### PR TITLE
Coord system not added to horizontal aux coords

### PR DESCRIPTION
**Pull request by dkillick**
Here's some changes to the NetCDF load rules that allow horizontal (specifically `latitude` and `longitude`) coordinates to be loaded with a coordinate system, if specified.

I've made no effort (yet) to get the tests passing as I'd like eyes on the proposed rules changes: are the changes _desirable_ or _sensible_? Given that I have a use case that requires the result (horizontal aux coords loaded from NetCDF with coord system intact), is there a better way of achieving this result?

### CODE REVIEW

**Comment by dkillick** [comment]
Reviewing #2026 introduced me to the simplification in the above commit. Still no updates so the tests pass, but I'm happier with the approach for what I need to achieve.

**Comment by dkillick** [comment]
:nudge:? @rhattersley @bjlittle @marqh @ajdawson do you have any comments on the approach/viability of this change?

File: lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb [hunk 1]
```diff
@@ -265,16 +265,21 @@ fc_build_auxiliary_coordinate_latitude
...
+        facts_cf.grid_mapping($grid_mapping)
+        check is_grid_mapping(engine, $grid_mapping, CF_GRID_MAPPING_LAT_LON)
     assert
         python cf_coord_var = engine.cf_var.cf_group.auxiliary_coordinates[$coordinate]
+        python cf_grid_var = engine.cf_var.cf_group.grid_mappings[$grid_mapping]
+        python coordinate_system = build_coordinate_system(cf_grid_var)
         python build_auxiliary_coordinate(engine, cf_coord_var,
-                                          coord_name=CF_VALUE_STD_NAME_LAT)
+                                          coord_name=CF_VALUE_STD_NAME_LAT,
+                                          coord_system=coordinate_system)
```
Comment (djkirkham) [inline lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb]: I'm not absolutely clear how Pyke works, but I'm wondering if the `foreach` will loop over every `$coordinate`/`$gridmapping` pair, creating duplicate coordinates with grid mappings with which they are not associated. I think there needs to be an extra check, something like `check is_coordinate_grid_mapping(engine, $coordinate, $grid_mapping)` which returns true if `$grid_mapping` is the correct grid mapping for `$coordinate`

File: lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb [hunk 2]
```diff
@@ -305,10 +310,15 @@ fc_build_auxiliary_coordinate_longitude
...
+        check is_grid_mapping(engine, $grid_mapping, CF_GRID_MAPPING_LAT_LON)
     assert
         python cf_coord_var = engine.cf_var.cf_group.auxiliary_coordinates[$coordinate]
+        python cf_grid_var = engine.cf_var.cf_group.grid_mappings[$grid_mapping]
+        python coordinate_system = build_coordinate_system(cf_grid_var)
         python build_auxiliary_coordinate(engine, cf_coord_var,
-                                          coord_name=CF_VALUE_STD_NAME_LON)
+                                          coord_name=CF_VALUE_STD_NAME_LON,
+                                          coord_system=coordinate_system)
         python engine.rule_triggered.add(rule.name)
```
Comment (djkirkham) [inline lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb]: See my comment abpove

Comment (djkirkham) [inline lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb]: See my comment above

File: lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb (hunk 1 above)
Comment (djkirkham) [inline lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb]: Never mind, I see this is an old commit

**Comment by marqh** [comment]
> Given that I have a use case that requires the result (horizontal aux coords loaded from NetCDF with coord system intact),

by implication, are Auxiliary Coordinates not given a CRS when loaded?
Is this endemic behaviour?

Could you give an example of a netCDF file header which loses the CRS information on load into Iris please?

Perhaps this could lead to a test case which fails, but which you think should pass, which would be a neat way of illustrating this problem in action

File: lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb [hunk 3]
```diff
@@ -268,13 +268,14 @@ fc_build_auxiliary_coordinate_latitude
     assert
         python cf_coord_var = engine.cf_var.cf_group.auxiliary_coordinates[$coordinate]
         python build_auxiliary_coordinate(engine, cf_coord_var,
-                                          coord_name=CF_VALUE_STD_NAME_LAT)
+                                          coord_name=CF_VALUE_STD_NAME_LAT,
+                                          coord_system=engine.provides['coordinate_system'])
```
Comment (pelson) [inline lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb]: How does the *correct* coordinate system find its way into the coordinate? I'd be concerned that we could get the one for the dimension coordinates...

**Comment by corinnebosley** [comment]
Closing because it sounds iffy and it's been stalled for a long time.

**Pull request closed without merging**